### ISSUE TITLE
Move music tracks & playlists code from Audio.cpp & Audio.h to new files (Music.cpp & Music.h?)

### ISSUE THREAD

**Issue by LeeSpork** [OpenLoco/OpenLoco#2763]
These music functions and array are more to do with specific game behaviour than the general audio system, so it would make sense to move them into a separate file. Might also make adding support for more custom music tracks in the future simpler.

@LeftofZen suggested "Playlist.cpp" or "Jukebox.cpp" in https://github.com/OpenLoco/OpenLoco/pull/2699#pullrequestreview-2441038783. My suggestion is simply "Music.cpp" to match what it is called in-game, although this name does have a caveat in that Audio.cpp would still retain some music-related functionality.

The following should be moved from Audio.cpp into this new file:
- `static constexpr MusicInfo kMusicInfo[]`
- `static std::vector<uint8_t> makeAllMusicPlaylist()`
- `static std::vector<uint8_t> makeCurrentEraPlaylist()`
- `static std::vector<uint8_t> makeCustomSelectionPlaylist()`
- `static int32_t chooseNextMusicTrack()`

And from from both Audio.cpp and Audio.h:
- `const MusicInfo* getMusicInfo(MusicId track)`
- `std::vector<uint8_t> makeSelectedPlaylist()`
- `void revalidateCurrentTrack()`

And from Audio.h:
- `struct MusicInfo`
- `using MusicId = uint8_t` (side note: this should be used in more places)
- `constexpr int32_t kNumMusicTracks = 29`

They should probably also be moved to a new namespace, such as `OpenLoco::Music`, instead of their current namespace `OpenLoco::Audio`.

**Comment by duncanspumpkin** [comment]
Agreed, audio is one area where we are fully implemented so can pretty much do what we want.

**Comment by LeeSpork** [comment]
Updated the list after remembering to also also consider Audio.h
